### PR TITLE
Permission handling added

### PR DESCRIPTION
**Pull request by ClaesNilsson**
- Updated section 4. Security and privacy considerations
- Added interface TCPPermission (UDPPermission and TCPserverPermission remains but wants to get feedback before going further)
- Editorial corrections
  https://github.com/sysapps/tcp-udp-sockets/issues/81

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -261,22 +269,195 @@
...
+             <dd>
+               The port of the peer.
+             </dd>                
+                             
+           </dl>      
+           
+         </dd>    
+
+         <dt>
+           Promise&lt;void&gt; requestPermission ()
```
Comment (jyasskin) [inline index.html:513]: What are your thoughts on requesting permission, and retrieving permission state for, multiple endpoints in a single call? Does it make sense for a site to be able to retrieve the complete set of endpoints it has access to?

File: index.html [hunk 2]
```diff
@@ -261,22 +269,195 @@
     <section>
       <h2>Security and privacy considerations</h2>
       <p>
-        This API must be only exposed to trusted content.
+        This is a security and privacy sensitive API and a <a>webapp</a> must
+        have permission to use the API. The manner in which permission is given 
+        or not varies depending on the type of web runtime environment in which 
+        this API is implemented and could be based on:
+        <ul>
+          <li>Consent through user interaction.</li>
```
Comment (jyasskin) [inline index.html:277]: Do you have any ideas on how to communicate the effect of granting access to various endpoints? What sorts of things should UAs consider including in their dialogs?

File: index.html [hunk 3]
```diff
@@ -261,22 +269,195 @@
...
+               <li>If permission was not granted, reject <code>tcpRequestPermissionPromise</code>
+                   with DOMException <code>"SecurityError"</code>. </li>  
+             </ol>      
+           </p>             
+           
+           <dl class='parameters'>
+               
+             <dt>DOMString remoteAddress</dt>
+             <dd>
+               The host name or IPv4/6 address of the peer.
```
Comment (jyasskin) [inline index.html]: The Chrome Apps sockets API includes wildcards in its permissions: https://developer.chrome.com/apps/app_network. Is that something you'd want to include here?

Comment (ClaesNilsson) [inline index.html]: Yes, it seems reasonable to have something like <host-pattern> here. Maybe this should be related to CSP connect-src?

File: index.html (hunk 1 above)
Comment (ClaesNilsson) [inline index.html:513]: See below on wildards.

File: index.html (hunk 2 above)
Comment (ClaesNilsson) [inline index.html:277]: See my comment at https://github.com/sysapps/tcp-udp-sockets/issues/81#issuecomment-66998129. Which is your general view on security/permissions for this API?

File: index.html (hunk 1 above)
Comment (jyasskin) [inline index.html:513]: Wildcards make it possible to request permission for multiple endpoints at once, but they don't allow retrieving the current set of allowed endpoints. This is a known hole in the Permissions API, which I'm wondering about (and haven't solved yet) for Web Bluetooth as well. I think Web Bluetooth will just provide a function to retrieve a sequence of already-allowed devices.

Obviously there's no need to add this function before merging this PR; it's just something to think about.

**Pull request merged**